### ISSUE TITLE
tmt should do /etc/hosts cleanup during the finish phase

### ISSUE THREAD

**Issue by kkaarreell** [teemtee/tmt#2102]
When running a multi-host test using tmt with `provision: connect` then tmt updates the system with some new records in `/etc/hosts`, however these updates are not removed when the  execution is over. 
This brings troubles when I am re-running the same plan against the same host (but different peers) multiple times.
Or, at least should not be adding new record if there is one present but rather update it.

**Comment by psss** [comment]
Yeah, looks reasonable, seems relevant for the `connect` provision plugin only, but doing the cleanup in general would not hurt, I guess.

**Comment by happz** [comment]
Hmmm, where does tmt update `/etc/hosts`? @kkaarreell can you provide an example of records added by tmt?

**Comment by kkaarreell** [comment]
I am afraid I can't. It is 2 years already. I have just tried to run a multi-host test with `provision: connect` and didn't encounter any updates in `/etc/hosts`. Feel free to close it, I am not using this approach frequently anyway and if I encounter issues I will report them with more details.

**Comment by happz** [comment]
I had to dive into `git log -p`, because it felt weird. But I think it was dropped in https://github.com/teemtee/tmt/pull/2072
